internal/infrastructure/kubernetes/ratelimit/resource_provider.go

```diff
@@ -54,8 +54,11 @@ func (r *ResourceRender) Service() (*corev1.Service, error) {
 	}
 
 	labels := rateLimitLabels()
+	if r.rateLimitDeployment.ServiceType == nil {
+		r.rateLimitDeployment.ServiceType = egcfgv1a1.DefaultKubernetesServiceType()
+	}
 
-	serviceSpec := resource.ExpectedServiceSpec(egcfgv1a1.DefaultKubernetesServiceType())
+	serviceSpec := resource.ExpectedServiceSpec(egcfgv1a1.GetKubernetesServiceType(*r.rateLimitDeployment.ServiceType))
```

Comment:
I dont think we need to define an API here, should be okay to hardcode to `Cluster`
the Rate Limit Service is internal to the Envoys living in the same cluster (and same namespace), so imo there is no need
to expose it to clients outside the cluster using `NodePort` or `LoadBalancer`

Reply:
This makes sense to me, I will write in the original issue that there will be just a change of the default setting.